File: app/src/main/java/com/vsa/filmoteca/presentation/utils/ConnectivityUtils.java
package com.vsa.filmoteca.presentation.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.vsa.filmoteca.FilmotecaApplication;

public class ConnectivityUtils {

    public static boolean isInternetAvailable(Context c) {
        ConnectivityManager cm = (ConnectivityManager) c.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = cm.getActiveNetworkInfo();
        // if no network is available networkInfo will be null, otherwise check if we are connected
        if (networkInfo != null && networkInfo.isConnected()) {
            return true;
        }
        return false;
    }

    public static boolean isInternetAvailable() {
        return isInternetAvailable(FilmotecaApplication.getInstance());
    }

}

File: app/src/main/java/com/vsa/filmoteca/presentation/utils/ConnectivityUtils.kt
package com.vsa.filmoteca.presentation.utils

import android.content.Context
import android.net.ConnectivityManager

object ConnectivityUtils {

    fun isInternetAvailable(c: Context): Boolean {
        val cm = c.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = cm.activeNetworkInfo
        // if no network is available networkInfo will be null, otherwise check if we are connected
        return networkInfo != null && networkInfo.isConnected
    }

}
